algolia/opt/composable_filter.go

```diff
@@ -86,13 +88,25 @@ func (o *composableFilterOption) UnmarshalJSON(data []byte) error {
 		ands = append(ands, ors)
 	}
 
-	if !ok && json.Unmarshal(data, &ors) == nil {
+	if !ok && json.Unmarshal(data, &options) == nil {
 		ok = true
-		ands = append(ands, ors)
-	}
 
-	if !ok && json.Unmarshal(data, &ands) == nil {
-		ok = true
+		for _, option := range options {
+			switch v := option.(type) {
+			case []interface{}:
+				ors = []string{}
+
+				for _, val := range v {
+					ors = append(ors, fmt.Sprint(val))
```

Comment:
Why is `fmt.Sprint` needed?

Reply:
Since we need to convert `[]interface{}` to `[]string`, I needed to convert `val`(interface{}) to string. Type conversion could also be done, but this approach seemed simpler to me.

Reply:
Oh, of course :ok_hand: